projects/igniteui-angular/src/lib/grids/grid/grid.pinning.spec.ts

```diff
@@ -1,1140 +1,681 @@
-            expect(headers[1].componentInstance.zIndex).toEqual(9998);
-
-            grid.pinColumn('Region');
-            tick();
-            fix.detectChanges();
-
-            // First three headers are pinned
-            headers = fix.debugElement.queryAll(By.directive(IgxGridHeaderGroupComponent));
-            expect(headers[2].componentInstance.zIndex).toEqual(9997);
-        }));
-
-        it('should not pin/unpin columns which are already pinned/unpinned', fakeAsync(() => {
-            const fix = TestBed.createComponent(DefaultGridComponent);
-            fix.detectChanges();
-
-            const grid = fix.componentInstance.instance;
-            const pinnedColumnsLength = grid.pinnedColumns.length;
-            const unpinnedColumnsLength = grid.unpinnedColumns.length;
+                expect(grid.getCellByColumn(0, currentColumn).value).toEqual(null);
+                expect(grid.getCellByColumn(0, releasedColumn).value).toEqual(true);
+                expect(grid.getCellByColumn(grid.data.length - 1, currentColumn).value).toEqual('Some other item with Script');
+                expect(grid.getCellByColumn(grid.data.length - 1, releasedColumn).value).toEqual(null);
+            });
+
+        });
 
-            let result = grid.pinColumn('CompanyName');
-            tick();
-            fix.detectChanges();
+        describe('', () => {
+            let fix;
+            let grid: IgxGridComponent;
 
-            expect(grid.pinnedColumns.length).toEqual(pinnedColumnsLength);
-            expect(result).toBe(false);
+            beforeEach(fakeAsync(() => {
+                fix = TestBed.createComponent(PinningComponent);
+                fix.detectChanges();
+                grid = fix.componentInstance.grid;
+            }));
 
-            result = grid.unpinColumn('City');
-            tick();
-            fix.detectChanges();
-
-            expect(grid.unpinnedColumns.length).toEqual(unpinnedColumnsLength);
-            expect(result).toBe(false);
-        }));
-
-        it('should not reject pinning a column if unpinned area width is less than 20% of the grid width', fakeAsync(() => {
-            const fix = TestBed.createComponent(GridPinningComponent);
-            const grid = fix.componentInstance.instance;
-            fix.detectChanges();
-            grid.columns.forEach((column) => {
-                if (column.index === 0 || column.index === 1 || column.index === 4 ||
-                    column.index === 6) {
-                    column.pinned = true;
-                }
+            it('should emit onColumnPinning event and allow changing the insertAtIndex param.', () => {
+
+                spyOn(grid.onColumnPinning, 'emit').and.callThrough();
+
+                let col = grid.getColumnByName('ID');
+                col.pinned = true;
+                fix.detectChanges();
+
+                expect(grid.onColumnPinning.emit).toHaveBeenCalledTimes(1);
```

Comment:
You have to add check for the event parameters:
expect(grid.onColumnPinning.emit).toHaveBeenCalledWith({});
Also have to added a check when unpin column.

Reply:
Done